airflow/models.py

```diff
@@ -2494,11 +2494,11 @@ def __init__(
         if args or kwargs:
             # TODO remove *args and **kwargs in Airflow 2.0
             warnings.warn(
-                'Invalid arguments were passed to {c}. Support for '
-                'passing such arguments will be dropped in Airflow 2.0. '
-                'Invalid arguments were:'
+                'Invalid arguments were passed to {c} (task_id: {t}). '
+                'Support for passing such arguments will be dropped in '
+                'Airflow 2.0. Invalid arguments were:'
                 '\n*args: {a}\n**kwargs: {k}'.format(
-                    c=self.__class__.__name__, a=args, k=kwargs),
+                    c=self.__class__.__name__, a=args, k=kwargs, t=task_id),
                 category=PendingDeprecationWarning
```

Comment:
Logging the task id sounds helpful, but if we add a parameter of `stacklevel=3` (I think 3 is the right value) then this will also include the filename and line number of the call. Could you add this in too?

(1 is this location, 2 is the constructor of the Operator itself, and 3 should be the line in the dag)